service/common/testing/fake.go

```diff
@@ -268,24 +264,25 @@ func (ss *FakeService) Remove() error {
 
 // InstallCommands implements Service.
 func (ss *FakeService) InstallCommands() ([]string, error) {
-	ss.AddCall("InstallCommands")
-
-	return nil, ss.NextErr()
+	return nil, ss.simpleCall("InstallCommands")
 }
 
 // StartCommands implements Service.
 func (ss *FakeService) StartCommands() ([]string, error) {
-	ss.AddCall("StartCommands")
-
-	return nil, ss.NextErr()
+	return nil, ss.simpleCall("StartCommands")
 }
 
 // WriteService implements UpgradableService.
 func (ss *FakeService) WriteService() error {
-	ss.AddCall("WriteService")
-	retErr := ss.NextErr()
-	if retErr != nil {
-		return retErr
-	}
-	return nil
+	return ss.simpleCall("WriteService")
+}
+
+// RemoveOldService implements UpgradableService.
+func (ss *FakeService) RemoveOldService() error {
```

Comment:
I would like to get rid of these fakes, but I already undertook some mechanical improvement work in this patch.